Fix missing vim genes

samuell opened the pull request:
# Description

Summary of the changes made:

This PR replaces the Spades assembler with Skesa, as it has been shown to produce generally higher quality assemblies and seems to work well with NovaSeq X data, thereby fixing multiple issues we have had with the microSALT pipeline.

The naming format used in the contigs fasta files are converted into the same format used by Spades immediately after assembly, so as not needing to change the scraping logic in microSALT.

Initial tests seem to show that assembly is also faster with Skesa.

## Primary function of PR
- [ ] Hotfix
- [x] Patch
- [ ] Minor functionality improvement
- [ ] New type of analysis
- [ ] Backward-breaking functionality improvement
- [ ] This change requires internal documents to be updated
- [ ] This change requires another repository to be updated

# Testing

_Test routine to verify the stability of the PR:_
- _`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh BRANCHNAME`_
- _`us`_
- _`source activate S_microSALT`_
- _(SITUATIONAL) `export MICROSALT_CONFIG=/home/proj/dropbox/microSALT.json`_
- _Select a relevant subset of the following:_
- _`microSALT analyse project MIC3109`_
- _`microSALT analyse project MIC4107`_
- _`microSALT analyse project MIC4109`_
- _`microSALT analyse project ACC5551`_

_Verify that the results for projects MIC3109, MIC4107, MIC4109 & ACC5551 are consistent with the results attached to AMSystem doc 1490, Microbial_WGS.xlsx_

## Test results
_These are the results of the tests, and necessary conclusions, that prove the stability of the PR._

# Sign-offs
- [ ] Code tested by @octocat
- [ ] Approved to run at Clinical-Genomics by @karlnyr

karlnyr reviewed (approved):
This looks great! Slight comment is that we have to update the implementation plan as we would need to update information in mlstfest:

https://github.com/search?q=repo%3AClinical-Genomics%2Fmlstfest%20spades&type=code

We send in assemblies, and we just have to change it so that we say it is SKESA and not spades

samuell commented:
To show that the VIM gene shows up after the fix, here is the resistances for the sample named above, for the `factualdodo` case:

In MicroSALT 3.3.5, where we can not find any VIM gene in this sample:

![image](https://github.com/user-attachments/assets/5039a7b9-e20c-4c1c-82fd-f655260da409)

In MicroSALT 4.0.0, though, we can see that blaVIM-4 shows up:

![image](https://github.com/user-attachments/assets/813b9785-ff81-4010-9b8e-fe5f4fec090b)

samuell commented:
As @karlnyr noted that here were some differences in the MLST calling for a couple of validation samples that he had run with both Spades with the --careful flag, and Skesa, below follows a walkthough and analysis of the differences.

The differences are created using the [htmldiff](https://github.com/seikichi/htmldiff) tool run on the MicroSALT Typing reports of the Spades/--careful and Skesa runs respectively. When there are differences, the Spades/--careful results will have a strikethrough effect, and the Skesa version will be underlined.

I have then gone through all the MLST sections in all the reports, and added screenshots of those that include differences, and added a comment to each:

The samples are:

- `ACC5551`
- `ACC155791
- `MIC3557`
- `MIC4109`

Find them below:

----

# ACC5551

## Case ...A9

![image](https://github.com/user-attachments/assets/0451affe-e873-42a9-8a0f-92e8a752343b)

**Comment:** Note that all the differing alleles had a non-100% length in the previous (Spades/--careful) run, and thus the differences can be explained by subpar quality in the Spades/--careful assembly.

[Being updated ...]
